pkg/agent/util/net_windows.go

```diff
@@ -613,92 +642,106 @@ func HostInterfaceExists(ifaceName string) bool {
 // there's no MTU field in HNSEndpoint:
 // https://github.com/Microsoft/hcsshim/blob/4a468a6f7ae547974bc32911395c51fb1862b7df/internal/hns/hnsendpoint.go#L12
 func SetInterfaceMTU(ifaceName string, mtu int) error {
-	cmd := fmt.Sprintf("Set-NetIPInterface -IncludeAllCompartments -InterfaceAlias \"%s\" -NlMtuBytes %d",
-		ifaceName, mtu)
-	_, err := runCommand(cmd)
-	return err
+	adapter, err := GetAdapterInAllCompartmentsByName(ifaceName)
+	if err != nil {
+		return fmt.Errorf("unable to find NetAdapter on host in all compartments with name %s: %v", ifaceName, err)
+	}
+	return adapter.SetMTU(mtu, antreasyscall.AF_INET)
 }
 
 func NewNetRoute(route *Route) error {
-	cmd := fmt.Sprintf("New-NetRoute -InterfaceIndex %v -DestinationPrefix %v -NextHop %v -RouteMetric %d -Verbose",
-		route.LinkIndex, route.DestinationSubnet.String(), route.GatewayAddress.String(), route.RouteMetric)
-	_, err := runCommand(cmd)
-	return err
+	if route == nil {
+		return nil
+	}
+	row := route.toMibIPForwardRow()
+	if err := antreaNetIO.CreateIPForwardEntry(row); err != nil {
+		return fmt.Errorf("failed to create new IPForward row: %v", err)
+	}
+	return nil
 }
 
 func RemoveNetRoute(route *Route) error {
-	cmd := fmt.Sprintf("Remove-NetRoute -InterfaceIndex %v -DestinationPrefix %v -Verbose -Confirm:$false",
-		route.LinkIndex, route.DestinationSubnet.String())
-	_, err := runCommand(cmd)
-	return err
-}
-
-func ReplaceNetRoute(route *Route) error {
-	rs, err := GetNetRoutes(route.LinkIndex, route.DestinationSubnet)
-	if err != nil {
-		return err
-	}
-
-	if len(rs) == 0 {
-		if err := NewNetRoute(route); err != nil {
-			return err
-		}
+	if route == nil || route.DestinationSubnet == nil {
 		return nil
 	}
-
-	for _, r := range rs {
-		if r.GatewayAddress.Equal(route.GatewayAddress) {
+	family := addressFamilyByIP(route.DestinationSubnet.IP)
+	rows, err := antreaNetIO.ListIPForwardRows(family)
+	if err != nil {
+		return fmt.Errorf("unable to list Windows IPForward rows: %v", err)
+	}
+	for i := range rows {
+		row := rows[i]
+		if row.DestinationPrefix.EqualsTo(route.DestinationSubnet) && row.Index == uint32(route.LinkIndex) && row.NextHop.IP().Equal(route.GatewayAddress) {
+			if err := antreaNetIO.DeleteIPForwardEntry(&row); err != nil {
+				return fmt.Errorf("failed to delete existing route %s: %v", route.String(), err)
+			}
 			return nil
```

Comment:
Is it expected to remove a single route only? Is it possible to have multiple routes matching the dest subnet and gateway address but different metrics?